### DISCUSSION TITLE
Get the tapped notification when multiple notifications are show

### DISCUSSION THREAD

**Discussion by AlexForget** [thudugala/Plugin.LocalNotification#359] [Q&A] [answered]
I will try to be clear as possible. 
I am testing the plugin and everything is really clear and simple to this point. But I have a problem with the NotificationActionTapped event. 

In my testing app I created two NotificationRequest with different NotificationId, Title, description, etc. I show the two notifications. The two notifications are send and I can tapped them. Their is notification 1 who is show first and notification 2 who is show second.

When I tapped notification 1 or 2, its always the data of the last show notification that its received. I cannot acces the data of the first show notification. 

Code exemple to help understand what I am trying to explain

I create the two notification

```
            var notification1 = new NotificationRequest
            {
                BadgeNumber = 1,
                Title = "Title 1",
                Description = "Description 1",
                ReturningData = "Data 1",
                NotificationId = 1,
                Schedule =
                {
                    NotifyTime = DateTime.Now.AddSeconds(5)
                },
            };

            var notification2 = new NotificationRequest
            {
                Android =
                {
                    Priority = AndroidPriority.High,
                },

                BadgeNumber = 1,
                Title = "Title 2",
                Description = "Description 2",
                ReturningData = "Data 2",
                NotificationId = 2,
                Schedule =
                {
                    NotifyTime = DateTime.Now.AddSeconds(5)
                },
            };

            LocalNotificationCenter.Current.Show(notification1);
            LocalNotificationCenter.Current.Show(notification2);
```
            
A little Display to show the received data after the tapped notification

```
        LocalNotificationCenter.Current.NotificationActionTapped += Current_NotificationActionTapped;

        private void Current_NotificationActionTapped(Plugin.LocalNotification.EventArgs.NotificationActionEventArgs e)
        {
            LocalNotificationCenter.Current.GetDeliveredNotificationList();

            App.Current.MainPage.DisplayAlert(e.Request.Title, e.Request.Description, "ok");
        }

```

The display is alway the data from the second notification

![image](https://user-images.githubusercontent.com/80180695/213459322-bd31f369-1e51-454f-b661-4c75b436a29b.png)

Is it by design or there is something I'm doing wrong?

Thank you!

**Comment by thudugala** [comment]
Can you attach a sample project please?

**Reply by AlexForget** [reply]
[XamarinLocalNotificationTest.zip](https://github.com/thudugala/Plugin.LocalNotification/files/10466983/XamarinLocalNotificationTest.zip)

Thank for your reply!

Here is a simple project. 

The expected behavior would be that once the two notifications are displayed in the header, when I press one or the other it is its information that is displayed in the popup window. At the moment, the information of the last notification that was sent is displayed.

Is this the expected behavior with your plugin? Or is there something I'm not doing right?

Thank again!!

**Comment by AlexForget** [comment]
I just tested the same test application with version 9.2 of the plugin and I get the desired behavior.

Hope this information can help.

**Comment by thudugala** [comment, marked as answer]
@AlexForget try version 10.1.4

**Reply by AlexForget** [reply]
Thanks, I will test it in the upcoming day!
